### PR TITLE
All setting arbitrary output resolutions for isoscape generation

### PR DESCRIPTION
**Pull request by VimWizardVersace**

### CODE REVIEW

File: generate_isoscape.ipynb
Comment (erickzul) [inline generate_isoscape.ipynb:1]: In "ter.animate(raster.load_raster(raster.get_raster_path("raster_test_old_res.tiff"), us" we should change this to the correct path.

**Review by erickzul — CHANGES_REQUESTED** [review]
Some clarification/confusion on what raster_size_x/y and whether it should match the resolution.

File: raster.py [hunk 1]
```diff
@@ -459,23 +459,28 @@ def d13C_var_geotiff() -> AmazonGeoTiff:
...
 }
 
+# Converts a resolution into a Bounds class, storing info about pixel size and raster size.
+def create_bounds_from_res(res: int, brazil_bounds: Bounds):
+  # Use brazil_bounds to get the min/max lat and lon. Scale everything else
+  # to fit the new resolution.
+  bounds = brazil_bounds
+  bounds.pixel_size_x *= (brazil_bounds.raster_size_x/res)
+  bounds.pixel_size_y *= (brazil_bounds.raster_size_y/res)
+  bounds.raster_size_x = res
```
[the remark this answers is not in the document]
Comment (VimWizardVersace) [reply raster.py:469]: raster_size_x and raster_size_y are the resolution, just pixel count on that axis. You're right that it is superfluous as you can infer the resolution from the min/max lat/long and the pixel size. Structured as is, however, the code to initialize new geotiffs depends on these members, so for that reason I choose to continue updating them.

File: raster.py [hunk 2]
```diff
@@ -459,23 +459,28 @@ def d13C_var_geotiff() -> AmazonGeoTiff:
...
-  output_resolution = sorted(
-    all_bounds,
-    key=lambda bounds: bounds.pixel_size_x*bounds.pixel_size_y)[-1 if max_res else 0]
-
-  # Randomly pick a medium sized raster.
-  if (not max_res):
-    output_resolution = get_extent(input_geotiffs["VPD"].gdal_dataset) 
+  
+  arbitrary_geotiff = list(input_geotiffs.values())[0]
+  brazil_bounds = get_extent(arbitrary_geotiff.gdal_dataset)
```
[the remark this answers is not in the document]
Comment (VimWizardVersace) [reply raster.py:482]: renamed to just "bounds"

File: generate_isoscape.ipynb
Comment (VimWizardVersace) [reply generate_isoscape.ipynb:1]: This shouldn't be here at all. Deleted

File: raster_test.ipynb
[the remark this answers is not in the document]
Comment (VimWizardVersace) [reply raster_test.ipynb:1]: Yes, raster_size_x/y is the width and height of the raster. If the resolution is 100x100 these should now be 100 and 100 and the pixel sizes should be scaled appropriately. That is what this test is checking.

File: raster.py [hunk 3]
```diff
@@ -459,23 +459,28 @@ def d13C_var_geotiff() -> AmazonGeoTiff:
   "ordinary_kriging_linear_d18O_predicted_variance" : krig_variances_isoscape_geotiff
 }
 
+# Converts a resolution into a Bounds class, storing info about pixel size and raster size.
+def create_bounds_from_res(res_x: int, res_y: int, brazil_bounds: Bounds):
+  # Use brazil_bounds to get the min/max lat and lon. Scale everything else
```
Comment (erickzul) [inline raster.py:464]: Do we want to change this from brazil_bounds to input_bounds instead?

File: raster_test.ipynb
[the remark this answers is not in the document]
Comment (erickzul) [reply raster_test.ipynb:1]: Could we use different values for width and height?

**Review by erickzul — CHANGES_REQUESTED** [review]

File: raster.py [hunk 4]
```diff
@@ -460,27 +460,28 @@ def d13C_var_geotiff() -> AmazonGeoTiff:
 }
 
 # Converts a resolution into a Bounds class, storing info about pixel size and raster size.
-def create_bounds_from_res(res_x: int, res_y: int, brazil_bounds: Bounds):
-  # Use brazil_bounds to get the min/max lat and lon. Scale everything else
+def create_bounds_from_res(res_x: int, res_y: int, bounds: Bounds):
```
Comment (erickzul) [inline raster.py:463]: Did you mean to have this parameter as 'base_bounds' like referenced throughout this function?

**Review by erickzul — CHANGES_REQUESTED** [review]

File: raster.py (hunk 3 above)
Comment (VimWizardVersace) [reply raster.py]: Yes

**Review by erickzul — APPROVED** [review]

**Pull request merged**